### ISSUE TITLE
false warning for 501

### ISSUE THREAD

**Issue by caleb15** [ansible-community/ansible-lint#679]
# Issue Type
- Bug report

# Ansible and Ansible Lint details

```
ansible-lint --version
ansible 2.7.10
  config file = /home/caleb/Documents/fifteen5-deploy/ansible.cfg
  configured module search path = ['/home/caleb/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/linuxbrew/.linuxbrew/opt/python/lib/python3.7/site-packages/ansible
  executable location = /home/linuxbrew/.linuxbrew/bin/ansible
  python version = 3.7.5 (default, Nov  1 2019, 19:15:52) [GCC 5.4.0 20160609]
ansible-lint 4.1.0
```

- ansible installation method: pip
- ansible-lint installation method: pip

# Desired Behaviour

No linting error

# Actual Behaviour (Bug report only)

Linting error 
`[501] become_user requires become to work as expected`

When a playbook has become set to true it applies to all tasks. 
https://docs.ansible.com/ansible/latest/user_guide/become.html#setting-enable-mode-for-all-tasks
Because of this it is not needed to set become at a task level.

**Comment by ssbarnea** [comment]
Closing because it is a good practice to be explicit as it would be very easy for introduce a regression if someone removes the become from the upper level. Still, if anyone wants to improve the rule and make it more intelligent, a PR will be welcomed.
